src/regexp.toit

```diff
@@ -1058,70 +1084,106 @@ class CharClass extends MiniExpAst:
       add 'z' + 1
           0xffff
 
-  case_insensitive_ranges old_ranges/List/*<int>*/ -> List/*<int>*/:
-    ranges/List/*<int>*/ := []
-    // TODO: Non-ascii.
-    letters := List 27: false  // 26 letters of alphabet plus a "known elephant in Cairo".
-    for i := 0; i < old_ranges.size; i += 2:
-      start/int := old_ranges[i]
-      end/int := old_ranges[i + 1]
-      if start <= 'z' and end >= 'A':
-        for j := start; j <= end; j++:
-          if 'A' <= j <= 'Z':
-            letters[j - 'A'] = true
-          else if 'a' <= j <= 'z':
-            letters[j - 'a'] = true
-    for i := 0; i < old_ranges.size; i += 2:
-      start/int := old_ranges[i]
-      end/int := old_ranges[i + 1]
-
-      // Any part before 'A':
-      e := min end 'A' - 1
-      if start <= e:
-        ranges.add start
-        ranges.add e
-      // Any part between 'Z' and 'a':
-      s := max start 'Z' + 1
-      e = min end 'a' - 1
-      if s <= e:
-        ranges.add s
-        ranges.add e
-      // Any part after 'z':
-      s = max start 'z' + 1
-      if s <= end:
-        ranges.add s
-        ranges.add end
-    start := -1
-    for i := 0; i < 27; i++:
-      if letters[i]:
-        if start == -1:
-          start = i
+  fix_ranges ranges/List/*<Range_>*/ case_sensitive/bool -> none:
```

Comment:
Add a comment explaining that this modifies the given $ranges list.

edit: Actually, since you do a copy just before calling this function, I would just return the result instead of modifying the parameter.